### PR TITLE
Remove headers fix

### PR DESCRIPTION
**Pull request by lewijw**
Please let me know if you see a problem.  We may want to wait for the Apache HTTPClient5 alpha2 release to happen before merging it into master.

### CODE REVIEW

File: .classpath [hunk 1]
```diff
@@ -42,8 +42,8 @@
 	<classpathentry kind="var" path="SAFS_BUILD/libs/autoit/AutoItX4Java.jar"/>
 	<classpathentry kind="var" path="SAFS_BUILD/libs/autoit/jacob.jar"/>
 	<classpathentry kind="var" path="SAFS_BUILD/libs/slf4j-api-1.7.21.jar"/>
-	<classpathentry kind="var" path="SAFS_BUILD/libs/httpclient5/httpclient5-5.0-alpha1.jar" sourcepath="/HTTPCLIENT5SOURCE"/>
-	<classpathentry kind="var" path="SAFS_BUILD/libs/httpclient5/httpcore5-5.0-alpha1.jar" sourcepath="/HTTPCORE_SOURCE"/>
+	<classpathentry exported="true" kind="var" path="SAFS_BUILD/libs/httpclient5/httpclient5-5.0-alpha2-SNAPSHOT.jar" sourcepath="/HTTPCLIENT5SOURCE"/>
+	<classpathentry exported="true" kind="var" path="SAFS_BUILD/libs/httpclient5/httpcore5-5.0-alpha2.jar" sourcepath="/HTTPCORE_SOURCE"/>
```
Comment (lewijw) [inline .classpath:46]: The httpclient and core jars are exported so they are on the classpath for test projects that depend on the project in Eclipse.

NOTE: the httpclient jar is a SNAPSHOT that I built.  We may want to wait for the official release before merging this into master.

File: src/org/apache/hc/client5/http/testframework/HttpClient5Adapter.java [hunk 2]
```diff
@@ -36,16 +36,16 @@
 import org.apache.hc.client5.http.impl.routing.DefaultProxyRoutePlanner;
 import org.apache.hc.client5.http.impl.sync.CloseableHttpClient;
 import org.apache.hc.client5.http.impl.sync.HttpClients;
-import org.apache.hc.client5.http.methods.CloseableHttpResponse;
+import org.apache.hc.client5.http.impl.sync.CloseableHttpResponse;
```
Comment (lewijw) [inline src/org/apache/hc/client5/http/testframework/HttpClient5Adapter.java:39]: Package names were changed between alpha1 and alpha2.

File: src/org/apache/hc/client5/http/testframework/HttpClient5Adapter.java [hunk 3]
```diff
@@ -134,7 +134,7 @@
         final String contentType = entity == null ? null : entity.getContentType();
 
         final Map<String, Object> ret = new HashMap<String, Object>();
-        ret.put("status", response.getStatusLine().getStatusCode());
+        ret.put("status", response.getCode());
```
Comment (lewijw) [inline src/org/apache/hc/client5/http/testframework/HttpClient5Adapter.java:137]: This is a change in API between alpha1 and alpha2.

File: src/org/safs/rest/service/Headers.java [hunk 4]
```diff
@@ -409,24 +412,26 @@ public static String convertHeadersMapToMultiLineString(Map<String,String> heade
...
+			final Header[] headers = AbstractMessageParser.parseHeaders(
 				inbuffer,
-				messageConstraints.getMaxHeaderCount(),
-				messageConstraints.getMaxLineLength(),
-				parser,
+				inputStream,
+				h1Config.getMaxHeaderCount(),
+				h1Config.getMaxLineLength(),
+				lineParser,
 				headerLines
```
Comment (lewijw) [inline src/org/safs/rest/service/Headers.java:435]: As you can see, there were a lot of changes between alpha1 and alpha2 in the way headers are parsed.

**Comment by twbutler** [comment]
So, it looks like what has been done is to simply upgrade to the new alpha2 version and make the necessary code changes required by the new version.  Looks OK.

**Comment by leiwang008** [comment]
Is this pull request still valid? We have not switched httpclient5 to alpha2 on master, but we are using the alpha2 on branch keyword_driven_safs_rest_testing. Shall we merge this to that branch?

**Comment by leiwang008** [comment]
I have verified that the commit https://github.com/SAFSDEV/Core/pull/32/commits/06c04943bd892eb7db041c636f2c9ebf03d4c7b6 is already in branch keyword_driven_safs_rest_testing. I will go ahead to close this pull request. If I made anything wrong, please let me know.

**Comment by lewijw** [comment]
I was just leaving it there in case someone wanted the headers fix without the recent changes.  After this long, I think it is safe to say that no one needs it.

**Pull request closed without merging**